Add IconTabLayout tests

arturdm opened the pull request:
This PR also adds a way to test Android library classes that use Android resources from other libraries.

arturdm committed:
Add IconTabLayout tests

tomrozb commented on lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java:
```diff
@@ -0,0 +1,92 @@
+  @Test public void shouldSetTabsFromPagerAdapterInstance() {
+    // given
+    IconTabLayout iconTabLayout = new IconTabLayout(RuntimeEnvironment.application);
+    given(pagerAdapter.getCount()).willReturn(2);
```
Is this line necessary?

arturdm replied:
Leftover after refactoring, I'll remove that.

arturdm committed:
Remove unnecessary mock method call

tomrozb commented on lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java:
```diff
@@ -0,0 +1,92 @@
+
+  @NonNull private AttributeSet createAttributeSetWithPageTitlesVisible(@NonNull Context context,
+      boolean pageTitlesVisible) {
+    return new RoboAttributeSet(Lists.newArrayList(
```
Didn't know assertj has Lists similar to Guava. Maybe I should switch to static factory methods in SignInButtonTest

arturdm replied:
Or maybe I should have refactored your code instead :).

tomrozb commented:
+1

arturdm committed:
Refactor SignInButtonTest to use assertj Lists

arturdm committed:
Refactor SignInButtonTest to remove duplicate code

arturdm committed:
Extract common code to AttributeSetUtils

arturdm committed:
Reorder tasks run by travis

If I put tasks connectedCheck and jacocoTestReport right
after another I get compilation errors.

arturdm commented:
If I run tasks in this order I get errors:

```
$ ./gradlew cC jacocoTestReport

...

:lib-android:compileDebugAndroidSupportDesign
error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider dagger.internal.codegen.ComponentProcessor not found
:lib-android:compileDebugAndroidSupportDesign FAILED
:lib-android:copyDebugAndroidSupportV7Appcompat UP-TO-DATE
:lib-android:compileDebugAndroidSupportV7Appcompat
error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider dagger.internal.codegen.ComponentProcessor not found
:lib-android:compileDebugAndroidSupportV7Appcompat FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lib-android:compileDebugAndroidSupportDesign'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lib-android:compileDebugAndroidSupportV7Appcompat'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

BUILD FAILED
```

arturdm committed:
Restore previous travis task order and provide fix

tomrozb commented:
+1 once again

tomrozb commented on lib-android/src/test/java/com/devinci/lib/AttributeSetUtils.java:
```diff
@@ -0,0 +1,30 @@
+import static org.assertj.core.util.Preconditions.checkNotNull;
+import static org.robolectric.Shadows.shadowOf;
+
+public final class AttributeSetUtils {
```
We should think about `lib-android-test-robolectric` module. This class  may be useful in other apps.

arturdm replied:
:+1:
